soda_machine.rb

```diff
@@ -1,18 +1,35 @@
+require_relative "soda.rb"
+
 class SodaMachine
   attr_reader :sodas, :cash
+  @@inventory = 0
 
   def initialize(args = {})
     @sodas = args[:sodas]
+    self.class.add_soda(@sodas.length)
     @cash = args[:cash]
   end
 
-  def current_inventory_count
+  def self.current_inventory_count
+    return @@inventory
```

Comment:
`@@inventory` is a class variable, not an instance variable.  What would happen if you had two separate soda machine instances?

Similarly, your `current_inventory_count` is a class method.  So you would have to call it on the class `SodaMachine`, not an instance of `SodaMachine`, which is what happens in the test.